template/Cargo.toml

```diff
@@ -92,6 +95,7 @@ embassy-time     = { version = "0.3.1",  features = ["generic-queue-8"] }
 esp-hal-embassy  = { version = "0.5.0",  features = ["esp32c6"] }
 static_cell      = { version = "2.1.0",  features = ["nightly"] }
 #ENDIF
+critical-section = "1.2.0"
```

Comment:
technically not needed but most users will end up needing it (partly resolves #51)